### PR TITLE
Jobradovic delete for category and formulas

### PR DESCRIPTION
**Pull request by jolpazy**

### CODE REVIEW

File: web/src/main/frontend/src/base/rest/requests.js [hunk 1]
```diff
@@ -2,17 +2,13 @@ import request from 'superagent';
...
 
-export const get = url => {
-  return request.get(getLink(url));
-};
+export const get = url => request.get(getLink(url));
 
-export const post = (url, data) => {
-  return request.post(getLink(url)).send(data);
-};
+export const post = url => request.post(getLink(url));
```
[the remark this answers is not in the document]
Comment (jolpazy) [reply web/src/main/frontend/src/base/rest/requests.js]: As it turns out, it's not even passed here, it is actually sent in formulaApi.js or wherever you use this post

File: web/src/main/frontend/src/components/Category.js [hunk 2]
```diff
@@ -70,6 +75,20 @@ class Category extends React.Component {
...
+                <div className="col-12">
+                  <div className="d-flex justify-content-end">
+                    {categoryForm.$('id').values() > 0 && (
+                      <Button
+                        click={() =>
+                          this.onDelete(categoryForm.$('id').values())
+                        }
+                      >
+                        {' '}
+                        Delete{' '}
```
[the remark this answers is not in the document]
Comment (jolpazy) [reply web/src/main/frontend/src/components/Category.js]: Don't know whats with the spaces, committing created them. definitely not for moving stuff around, ill try and remove 'em

File: web/src/main/frontend/src/components/Formula.js [hunk 3]
```diff
@@ -89,6 +94,20 @@ class Formula extends React.Component {
                 <div className="col-12 add__button_container">
                   <Button cssClass="icon__block" title="+" />
                 </div>
+                <div className="col-12">
+                  <div className="d-flex justify-content-end">
+                    {formulaForm.$('id').values() > 0 && (
+                      <Button
+                        click={() =>
```
[the remark this answers is not in the document]
Comment (jolpazy) [reply web/src/main/frontend/src/components/Formula.js]: Oki

File: web/src/main/frontend/src/components/Category.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sugrinov) [reply web/src/main/frontend/src/components/Category.js]: :trollface:

File: web/src/main/frontend/src/components/Category.js [hunk 4]
```diff
@@ -17,6 +17,11 @@ class Category extends React.Component {
     categoryForm.update(selectedCategory);
   };
 
+  onDelete(id) {
+    this.props.category.deleteCategory(id);
+    categoryForm.clear();
```
[the remark this answers is not in the document]
Comment (jolpazy) [reply web/src/main/frontend/src/components/Category.js]: onReset and onClear break the app for some reason...

**Pull request merged**